### PR TITLE
Enable viewport scaling, fix view assumptions

### LINKED ISSUE #1612: add WebXR dynamic viewport scaling support
### Description

Feature request: add support for dynamic viewport scaling as recently added to the WebXR spec. Since this isn't widely supported in user agents yet, make it conditional on support being present.

The WebXR explainer section https://github.com/immersive-web/webxr/blob/master/explainer.md#dynamic-viewport-scaling has documentation for the feature.

### PR DESCRIPTION
**Pull request by klausw**
Use dynamic viewport scaling using the recommended viewport scale
if supported.

For correct rendering, loop over the returned view array instead of just 
hardcoding the first view, and use each view's corresponding viewport.

### Reference Issue
Fixes #1612

### CODE REVIEW

File: packages/model-viewer/src/three-components/ARRenderer.ts [hunk 1]
```diff
@@ -457,6 +460,15 @@ export class ARRenderer extends EventDispatcher {
       this.dispatchEvent({type: 'status', status: ARStatus.SESSION_STARTED});
     }
 
+    // Use automatic dynamic viewport scaling if supported.
+    if (view.requestViewportScale && view.recommendedViewportScale) {
+      const scale = view.recommendedViewportScale;
+      view.requestViewportScale(Math.max(scale, MIN_VIEWPORT_SCALE));
+    }
+    const layer = this[$currentSession]!.renderState.baseLayer;
+    const viewport = layer!.getViewport(view);
```
Comment (elalish) [inline packages/model-viewer/src/three-components/ARRenderer.ts:469]: Our tests are failing on this line in Chrome 85. Any idea why?

**Review by elalish — CHANGES_REQUESTED** [review]

File: packages/model-viewer/src/three-components/ARRenderer.ts (hunk 1 above)
Comment (klausw) [reply packages/model-viewer/src/three-components/ARRenderer.ts:469]: The mock XRSession data was missing the getViewport method on the layer.

Is there a way to run tests individually? My laptop seems to have trouble running the full suite, so the fix is a bit speculative.

Comment (elalish) [reply packages/model-viewer/src/three-components/ARRenderer.ts:469]: Yes, on any of the `test()` or `suite()` calls, just add a `.only`, like `test.only()`.

Comment (elalish) [reply packages/model-viewer/src/three-components/ARRenderer.ts:469]: Anyway, it passed our CI, so looks like it's all good. Thanks!

**Review by elalish — APPROVED** [review]
For some reason my Chrome Canary refuses to update, even though Play knows it's out of date (still on 85 for some reason). This is working fine on 85, so I'll just merge it once the tests pass and let's test it on the new thing.

**Pull request merged**